### PR TITLE
Edit mutation

### PR DESCRIPTION
**Pull request by cbuchko**
- Editing the table/calendar now flags the scheduler into an "editing" mode.
- Clicking "submit changes" pushes all unsaved changes to the backend through the mutation
- Added select input for courses
- Haven't implemented validation skipping or choosing which company does validation. Gonna do this in a separate PR

### CODE REVIEW

File: src/components/Schedule/AppointmentModal.tsx
Comment (jayden-chan) [inline src/components/Schedule/AppointmentModal.tsx:68]: I think you can do this
```suggestion
  value: CourseID;
```

File: src/components/Schedule/AppointmentModal.tsx
Comment (jayden-chan) [inline src/components/Schedule/AppointmentModal.tsx]: Would this be an error that we want to surface to the user or just ignore? Either way we should probably clean up the console log before merging

File: src/components/Schedule/SubmitButton.tsx
Comment (jayden-chan) [inline src/components/Schedule/SubmitButton.tsx]: Put a `duration: null` on any error modals so they don't disappear and the user has time to read what went wrong

File: src/pages/Schedule.tsx
Comment (jayden-chan) [inline src/pages/Schedule.tsx:130]: Same thing here, does this error need to be displayed in a toast to the user or can we safely ignore it?

File: src/components/Schedule/SubmitButton.tsx
Comment (jayden-chan) [inline src/components/Schedule/SubmitButton.tsx]: I think we probably want to wait to update this until the mutation succeeds no?

File: src/components/Schedule/SubmitButton.tsx
Comment (jayden-chan) [inline src/components/Schedule/SubmitButton.tsx:31]: Kind of too late for this but the coupling of state here between this component and the parent is a bit scuffed. Normally you would either do all the state management in this component (don't pass in `active` and `handleSubmit`) or do everything in the parent component (do the mutation inside of `handleSubmit` instead of in this component). Not worth changing anything if this works just a comment.

**Review by jayden-chan — APPROVED** [review]
Looks good just some minor comments

File: src/components/Schedule/AppointmentModal.tsx
Comment (jayden-chan) [reply src/components/Schedule/AppointmentModal.tsx:68]: or is CourseID the one with `term` lol I can't remember

**Review by AndrewPolanyi — APPROVED** [review]
great work connor! a bit over my head but approving so that your merge isnt blocked

**Review by nicudy — APPROVED** [review]
Looks great, Connor!

File: src/components/Schedule/SubmitButton.tsx
Comment (cbuchko) [reply src/components/Schedule/SubmitButton.tsx:31]: yeah definitely scuffed lol

File: src/components/Schedule/AppointmentModal.tsx
Comment (cbuchko) [reply src/components/Schedule/AppointmentModal.tsx:68]: yeah CourseID includes term, and we don't support editing of terms so I left it out

**Pull request merged**
